File: src/main/Python/boj_step10/boj5073.py
import sys
cin = sys.stdin.readline

while True:
    a, b, c = map(int, cin().split())
    if a == b == c == 0: exit()
    M = max(a, max(b, c))
    if M * 2 >= a + b + c: print('Invalid')
    else:
        if a == b == c: print('Equilateral')
        elif a == b or b == c or a == c: print('Isosceles')
        else: print('Scalene')

File: src/main/Cpp/boj_step10/boj5073.cpp
#include <iostream>
using namespace std;

int main() {
    while (true) {
        int a, b, c;
        cin >> a >> b >> c;

        if (a == b && b == c && a == 0) return 0;

        int max_value = max(a, max(b, c));

        if (max_value * 2 >= a + b + c)
            cout << "Invalid";

        else {
            if (a == b && b == c)
                cout << "Equilateral";

            else if (a == b || b == c || a == c)
                cout << "Isosceles";

            else
                cout << "Scalene";
        }
    }
}